programs/mango-v4/src/instructions/serum3_edit_market.rs

```diff
@@ -0,0 +1,25 @@
+use anchor_lang::prelude::*;
+
+use crate::state::*;
+
+#[derive(Accounts)]
+#[instruction(market_index: Serum3MarketIndex)]
+pub struct Serum3EditMarket<'info> {
+    #[account(
+        has_one = admin,
+        constraint = group.load()?.serum3_supported()
+    )]
+    pub group: AccountLoader<'info, Group>,
+    pub admin: Signer<'info>,
+
+    #[account(
+        mut,
+        has_one = group
+    )]
+    pub market: AccountLoader<'info, Serum3Market>,
+}
+
+pub fn serum3_edit_market(ctx: Context<Serum3EditMarket>, reduce_only: bool) -> Result<()> {
```

Comment:
since this is not unlikely to expand in the future, use the usual *_edit pattern where we pass Options

Reply:
done, please check